odoo/addons/base/models/ir_mail_server.py

```diff
@@ -190,7 +197,7 @@ def test_smtp_connection(self):
         }
 
     def connect(self, host=None, port=None, user=None, password=None, encryption=None,
-                smtp_debug=False, mail_server_id=None, ssl_certificate=None, ssl_private_key=None):
+                smtp_debug=False, mail_server_id=None, smtp_from=None, ssl_certificate=None, ssl_private_key=None):
```

Comment:
Side remark: somehow with this PR connect is becoming a big if mail_server / else with more and more divergent computation. Couldn't we think of moving mail_server-based connect into a ``connect_with_mail_server``, as it would simplify its calls ? Manual connect could be ``connect_manual`` (or something like that). ``connect`` would be deprecated but would redirect to one or the other one, depending on given parameters, for backward compatibility. What do you think ?

Reply:
Not sure it's a good idea, because there're a lot of code after the big if / else